integration/push_test.go

```diff
@@ -194,3 +195,122 @@ func TestLegacyOCI(t *testing.T) {
 		})
 	}
 }
+
+func TestPushWithExistingIndices(t *testing.T) {
+	t.Parallel()
+	regConfig := newRegistryConfig()
+	sh, done := newShellWithRegistry(t, regConfig)
+	defer done()
+
+	rebootContainerd(t, sh, getContainerdConfigToml(t, false), getSnapshotterConfigToml(t, false))
+	images := []string{nginxImage, rabbitmqImage, drupalImage, alpineImage}
+
+	imageToIndexDigest := make(map[string]string)
+	imageToManifestDigest := make(map[string]string)
+	for _, img := range images {
+		mirrorImg := regConfig.mirror(img)
+		copyImage(sh, dockerhub(img), mirrorImg)
+		indexDigest := buildIndex(sh, mirrorImg, withSpanSize(1<<22), withOCIArtifactRegistrySupport)
```

Comment:
They are since the default registry config has `oci11RegistryImage`.
https://github.com/awslabs/soci-snapshotter/blob/00474ecf61cfb905a0d6436b516110050bb6435c/integration/testutil.go#L413-L418


 I guess that should be changed  to the 1.0 registry. I'll make that change with this PR

Reply:
I think it should be fine to use that registry, since even OCI 1.1 registry should also support index as image-manifest right?

Reply:
True its the other way around that isnt supported

Reply:
Then we can just keep the oci 1.1 registry, which supports both artifact and image. It's no hurt anyway. If artifact is really removed in oci 1.1 or when we decide to remove artifact, we can remove this oci 1.1 registry. 

This actually reminds me, why do we need to have [both oci 1.1 and oci 1.0 registry](https://github.com/awslabs/soci-snapshotter/blob/00474ecf61cfb905a0d6436b516110050bb6435c/integration/testutil.go#L74-L75)? If oci 1.1 registry support both image|artifact🤔 The only scenario I see is to test `artifact` cannot be pushed to a oci 1.0 registry, but I wonder if such a test scenario makes much sense...

Reply:
Yeah thats the only case in which we need it. I mean I think Azure may have a concept of artifacts so there may still be users who will be interacting with OCI 1.1 registries... Not sure how long that will last though

Reply:
yeah I mean given currently we support both artifact/image and the oci 1.1 registry in our test also supports both, I didn't see a practical testcase where we really need a oci 1.0 (i.e. image-support only) registry other than the mentioned case.

Reply:
The OCI 1.1 registry also introduced the referrers API, so we're taking a different code path to retrieve the SOCI index depending on which repository we're using. Arguably this is really just testing that `oras-go` does the right thing so maybe we can skip it.

Reply:
I'm fine with that I suppose that can be apart of another PR (cleaning up OCI 1.0 img references + removing it from our Dockerfile)